packages/lesswrong/components/languageModels/LlmConversationsViewingPage.tsx

```diff
@@ -0,0 +1,227 @@
+      <span className={classes.conversationRowTitle}>{title}</span>
+    </span>
+    <span className={classes.conversationRowNumLastUpdated}><Components.FormatDate date={lastUpdatedAt ?? createdAt}/></span>
+  </div>
+}
+
+
+// Lists all conversations for admins to select from
+const LlmConversationSelector = ({currentConversationId, setCurrentConversationId, classes}: {
+  currentConversationId: string|undefined,
+  setCurrentConversationId: (conversationId: string) => void,
+  classes: ClassesType<typeof styles>,
+}) => {
+
+  const { results, loading } = useMulti({
+    collectionName: "LlmConversations",
+    fragmentName: "LlmConversationsWithUserInfoFragment",
+    terms: { view: "llmConversationsAll" },
+  });
+
+  useEffect(() => {
+    if (results) {
+      setCurrentConversationId(results[0]?._id);
+    }
+    //eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!results && loading) {
+    return <Components.Loading />
+  }
+  
+  if (!results) {
+    return <div>No conversations found</div>;
+  }
+
+  return <div className={classes.conversationSelectorRoot}>
+    {results.map((conversation, idx) => {
+      return <LlmConversationRow
+      key={idx} 
+      conversation={conversation}
+      currentConversationId={currentConversationId}
+      setCurrentConversationId={setCurrentConversationId}
+      classes={classes} />;
+    })}
+  </div>
+}
+
+
+const LlmConversationViewer = ({conversationId, classes}: {
+  conversationId: string|undefined,
+  classes: ClassesType<typeof styles>,
+}) => {
+
+  const { LlmChatMessage, SectionTitle } = Components
+
+  const { document: conversation, loading } = useSingle({
+    collectionName: "LlmConversations",
+    fragmentName: "LlmConversationsWithMessagesFragment",
+    documentId: conversationId,
+    skip: !conversationId
+  });
+
+
+  if (!conversationId) {
+    return <div className={classes.conversationViewer}>
+      No conversation selected
+    </div>
+  }
+
+  if (!conversation && loading) {
+    return <div className={classes.conversationViewer}>
+      <Components.Loading />
+      </div>
+  }
+
+  if (!conversation) {
+    throw new Error("No conversation with selectedId");
+  }
+  
+  const { title, messages } = conversation;
+
+  return <div className={classes.conversationViewer}>
+    <SectionTitle title={title} titleClassName={classes.conversationViewerTitle} />
+    {!messages?.length && <div>No messages in this conversation</div>}
+    {messages?.length && messages.map((message, idx) => {
+      return <LlmChatMessage  key={idx} message={message} />
+    })}
+  </div> 
+}
+
+
+export const LlmConversationsViewingPage = ({classes}: {
+  classes: ClassesType<typeof styles>,
+}) => {
+  const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+
+  const currentUser = useCurrentUser();
+  const [currentConversationId, setCurrentConversationId] = React.useState<string|undefined>(undefined);
```

Comment:
you don't need to explicitly set `undefined` in the generic type, you can just do `useState<string>()` and it automatically turns it into `string | undefined`.  (You also don't need to explicitly pass `undefined` in to the initial value in that case, either.)

Reply:
Also prefer to import `useState` rather than doing `React.useState`